ops/terraform/services/pipeline/deploy.groovy

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env groovy
+
+// TODO: Consider further generalizing this and hoisting into a new ops/jenkins/global-pipeline-libs/terraform.groovy
+
+/* Deploys base terraservice
+ * @param args a {@link Map} must include `env`, `directory`
+ * <ul>
+ * <li>env string represents the targeted BFD SDLC Environment
+ * <li>directory string relative path to terraservice module directory
+ * <li>tfVars optional map represents module's terraform input variables and their respective values
+ * </ul>
+*/
+void deployTerraservice(Map args = [:]) {
+    bfdEnv = args.env
+    serviceDirectory = args.directory
+    tfVars = args.tfVars ?: [:]
+
+    // format terraform variables
+    tfVarsCollection = []
+    tfVars.each { key, val -> tfVarsCollection += "\"-var=$key=$val\""}
+    terraformVariables = tfVarsCollection.join(" ")
```

Comment:
You can shorten this down to something like the following:

```groovy
terraformVariables = tfVars.collect { key, val -> "\"-var=$key=$val\"" }.join(" ")
```

Reply:
Thank, @malessi. That's cleanear.